### PR TITLE
mariadb: remove development components

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Remove development components. Need to use only libmysqlclient package.
Update build configuration.

Please recheck build configuration.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by Izorkin** [comment]
@GrahamcOfBorg build zoneminder

**Comment by Izorkin** [comment]
@GrahamcOfBorg build perlPackages.SysCPU

**Comment by volth** [comment]
> SysCPU

There is a bugfix for aarch64: https://github.com/NixOS/nixpkgs/pull/64748#issuecomment-516752285

**Comment by Izorkin** [comment]
@GrahamcOfBorg build perlPackages.SysCPU

**Comment by Izorkin** [comment]
@volth thanks, added patch.

**Comment by Izorkin** [comment]
@GrahamcOfBorg build perlPackages.SysCPU

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -16332,6 +16332,12 @@ let
       url = mirror://cpan/authors/id/M/MZ/MZSANFORD/Sys-CPU-0.61.tar.gz;
       sha256 = "1r6976bs86j7zp51m5vh42xlyah951jgdlkimv202413kjvqc2i5";
     };
+    patches = [
+      # Bug #95400 for Sys-CPU: Tests fail on ARM and AArch64 Linux
+      # https://rt.cpan.org/Public/Bug/Display.html?id=95400
+      ../development/perl-modules/perl-SysCPU-Add-support-for-cpu_type-on-ARM-and-AArch64-Linux-pl.patch
```
Comment (Mic92) [inline pkgs/top-level/perl-packages.nix]: fetchpatch is preferred here. You can use: https://rt.cpan.org/Ticket/Attachment/1359669/721669/0001-Add-support-for-cpu_type-on-ARM-and-AArch64-Linux-pl.patch

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -16332,6 +16332,12 @@ let
...
+      ../development/perl-modules/perl-SysCPU-cpu_clock-can-be-undefined-on-an-ARM.patch
```
Comment (Mic92) [inline pkgs/top-level/perl-packages.nix]: Same:

https://rt.cpan.org/Ticket/Attachment/1388036/737125/0002-cpu_clock-can-be-undefined-on-an-ARM.patch

**Comment by volth** [comment]
it is already fixed in staging by `doCheck = !stdenv.isAarch64`


The patch does nothing besides that

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (Izorkin) [reply pkgs/top-level/perl-packages.nix]: Thanks, fixed.

**Comment by Izorkin** [comment]
Use fetchpatch or doCheck ?

**Comment by Mic92** [comment]
If tests work fine after applying patches, we can drop `doCheck`.

**Comment by Izorkin** [comment]
@GrahamcOfBorg build perlPackages.SysCPU

**Comment by Izorkin** [comment]
@GrahamcOfBorg build perlPackages.SysCPU

**Comment by Izorkin** [comment]
@GrahamcOfBorg build mariadb zoneminder

**Comment by Izorkin** [comment]
Update Mariadb to version 10.3.19

**Comment by Izorkin** [comment]
Update package to version 10.3.20

сс @thoughtpolice @matthewbauer @Mic92

**Pull request merged**